### ISSUE TITLE
Can't spawn a container with --read-only under debian testing

### ISSUE THREAD

**Issue by amigrave** [systemd/systemd#5170]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### systemd version the issue has been seen with

> $ systemd --version
systemd 232
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN

*NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!*

### Used distribution

> Debian testing (also tried on debian sid too).

### In case of bug report: Expected behaviour you didn't see

> being able to spawn a container using `--read-only`

### In case of bug report: Unexpected behaviour you saw

> $ systemd-nspawn -D /var/lib/machines/test --read-only                    
```
Spawning container test on /var/lib/machines/test.
Press ^] three times within 1s to kill container.
Failed to create directory /var/lib/machines/test/sys: Read-only file system
```

### In case of bug report: Steps to reproduce the problem

> $ debootstrap --arch amd64 xenial /var/lib/machines/test http://archive.ubuntu.com/ubuntu
$ systemd-nspawn -D /var/lib/machines/test --read-only

**Comment by amigrave** [comment]
Here is a screencast showing the problem just in case:
[![asciicast](https://asciinema.org/a/0on4hm1f8eypolm2mqnbv2ass.png)](https://asciinema.org/a/0on4hm1f8eypolm2mqnbv2ass)

**Comment by fsateler** [comment]
My best guess is that this happens because nspawn tries to do the uid shift[1], and that can't work when the filesystem is read-only.

This does not happen in 231, so I mark it as a regression


[1] https://github.com/systemd/systemd/blob/master/src/nspawn/nspawn-mount.c#L489

**Comment by evverx** [comment]
> This does not happen in 231

Try to remove `<container_root>/tmp`

Actually, [`--read-only`](https://github.com/systemd/systemd/commit/bc2f673ec24b59948fcfc35b) doesn't work. You should create:
```
sys
dev
run
tmp
proc
var/log (depends on --link-journal usage)
```
manually

**Comment by fsateler** [comment]
@evverx I'm confused, why did you remove the regression label? The regression is that a directory tree that contains `/sys` dir is not startable in readonly mode.

**Comment by evverx** [comment]
@fsateler , sorry. You're right. This is the regression.
The fix is acbbf69b718260755a

**Comment by fsateler** [comment]
@evverx ok, so this is already fixed. Excellent! Closing, as this was a duplicate of #4711
